### PR TITLE
Fix Nmap XML parser for tunnel attribute

### LINKED ISSUE #3821: nmap parser doesn't parse "SSL/HTTPS?" output
*This issue was [RM8805](https://dev.metasploit.com/redmine/issues/8805), filed by @wchen-r7 on behalf of @sho-luv*

The following nmap xml example shows a Apache web service running on port 443 with SSL, but the name parameter doesn't say "https", it says "http"

```xml
<port protocol="tcp" portid="443"><state state="open" reason="syn-ack" reason_ttl="52"/><service name="http" product="Apache httpd" tunnel="ssl" method="probed" conf ="10"><cpe>cpe:/a:apache:http_server</cpe></service></port>
```

If you import that to Metasploit it'll be treated as HTTP and not HTTPS.

### PR DESCRIPTION
**Pull request by wvu**
Fixes #3821.

### CODE REVIEW

**Comment by wvu-r7** [comment]
GitHub connection problems failed the last build.

File: lib/msf/core/db_manager/import/nmap.rb [hunk 1]
```diff
@@ -164,9 +164,12 @@ def import_nmap_xml(args={}, &block)
         data[:host]  = hobj || addr
         data[:info]  = extra if not extra.empty?
         data[:task]  = args[:task]
-        if p["name"] != "unknown"
+        if p["name"]
           data[:name] = p["name"]
         end
+        if p['tunnel']
+          data[:name] = "#{p['tunnel']}/#{data['name'] || 'unknown'}"
```
Comment (bcook-r7) [inline lib/msf/core/db_manager/import/nmap.rb]: should this be #{p['name']}, not #{data['name']} ?

Comment (wvu-r7) [inline lib/msf/core/db_manager/import/nmap.rb]: It can be. This parser's different, so either will work. Lemme consolidate into the previous check. :)

Comment (bcook-r7) [inline lib/msf/core/db_manager/import/nmap.rb]: Sorry, didn't mean to bikeshed over one silly line :)

Comment (wvu-r7) [inline lib/msf/core/db_manager/import/nmap.rb]: Fixed for more better!

**Comment by bcook-r7** [comment]
Looks good to me:
```
before: 192.168.56.1  443   tcp    http      open   Apache httpd
after:  192.168.56.1  443   tcp    ssl/http  open   Apache httpd
```

**Comment by wvu-r7** [comment]
Thanks!!

**Pull request merged**